### PR TITLE
Update ssh task lib

### PR DESCRIPTION
**Pull request by damccorm**

### CODE REVIEW

**Comment by damccorm** [comment]
This one took some more work than the others because it had legacy tests using an old framework depending on the old task-lib behavior that needed to be upgraded/rewritten. That's why the change set is bigger

File: Tasks/SshV0/Tests/L0.ts [hunk 1]
```diff
@@ -1,16 +1,170 @@
...
+        tr.run();
+
+        runValidations(() => {
+            assert(tr.invokedToolCount == 0, 'should not have run any tools');
+            assert(tr.failed, 'task should have failed');
+            assert(tr.stdout.indexOf('Endpoint auth data not present: IDUserNameNotSet') >= 0, 'wrong error message: "' + tr.stdout + '"');
+        }, tr, done);
     });
 
-    it('Does a basic hello world test', function(done: MochaDone) {
```
Comment (jtpetty) [inline Tasks/SshV0/Tests/L0.ts]: did we lose this test? Was it not doing anything worthwhile?

File: Tests-Legacy/L0/SshV0/_suite.ts [hunk 2]
```diff
@@ -1,202 +0,0 @@
...
-                assert(tr.stderr.length > 0, 'should have written to stderr');
-                assert(tr.failed, 'task should have failed');
-                assert(tr.stderr.indexOf('Input required: sshEndpoint') >= 0, 'wrong error message: "' + tr.stderr + '"');
-                done();
-            })
-            .fail((err) => {
-                done(err);
-            });
-    })
-    //it('Fails for invalid private key', (done) => {
```
Comment (jtpetty) [inline Tests-Legacy/L0/SshV0/_suite.ts]: this was commented out, but should it have been?

File: Tasks/SshV0/Tests/L0.ts (hunk 1 above)
Comment (damccorm) [reply Tasks/SshV0/Tests/L0.ts]: Yeah, it was a placeholder

File: Tests-Legacy/L0/SshV0/_suite.ts (hunk 2 above)
Comment (damccorm) [reply Tests-Legacy/L0/SshV0/_suite.ts]: I'm not 100% sure, but its been commented out for the past 3 years, maybe longer. I think its covered by the tests further down (specifically 'Fails when connection cannot be made with given details')

**Pull request merged**